mne_bids/copyfiles.py

```diff
@@ -339,6 +341,132 @@ def copyfile_brainvision(vhdr_src, vhdr_dest, anonymize=None, verbose=False):
+            will always be anonymized if present, regardless of this setting.
+
+    See Also
+    --------
+    mne.io.anonymize_info
+
+    """
+    # Ensure source & destination extensions are the same
+    fname_src, ext_src = _parse_ext(src)
+    fname_dest, ext_dest = _parse_ext(dest)
+    if ext_src != ext_dest:
+        raise ValueError(f'Need to move data with same extension'
+                         f' but got "{ext_src}", "{ext_dest}"')
+
+    # Copy data prior to any anonymization
+    sh.copyfile(src, dest)
+
+    # Anonymize EDF/BDF data, if requested
+    if anonymize is not None:
+        if ext_src == '.bdf':
+            raw = read_raw_bdf(dest, preload=False, verbose=0)
+        else:
+            raw = read_raw_edf(dest, preload=False, verbose=0)
+
+        # Get patient info, recording info, and recording date
+        with open(dest, 'rb') as f:
+            f.seek(8)  # id_info field starts 8 bytes in
+            id_info = f.read(80).decode('ascii').rstrip()
+            rec_info = f.read(80).decode('ascii').rstrip()
+
+        # Parse metadata from file
+        if len(id_info) == 0 or len(id_info.split(' ')) != 4:
+            id_info = "X X X X"
+        if len(rec_info) == 0 or len(rec_info.split(' ')) != 5:
+            rec_info = "Startdate X X X X"
+        pid, sex, birthdate, name = id_info.split(' ')
+        startdate, admin_code, tech, equip = rec_info.split(' ')[1:5]
+
+        # Try to anonymize recording date using 4-digit year from
+        # "Startdate" field from "local recording identification"
+        # section, because the generic "startdate" field only
+        # supports 2-digit years.
+        #
+        # XXX: recording dates are specified twice in EDF/EDF+/BDF,
+        # MNE-Python currently parses the 2-digit year field. If that is
+        # changed to defaulting to the 4-digit field, this try-except
+        # clause can be removed. See:
+        # https://github.com/mne-tools/mne-python/issues/8544
+        try:
+            real_year = datetime.strptime(startdate, "%d-%b-%Y").year
+            newdate = raw.info['meas_date'].replace(year=real_year)
+            raw.info['meas_date'] = newdate
+        except ValueError as e:
+            # We could not parse the "Startdate" field from "local recording
+            # identification" section (e.g., because it was "X").
+            # So fall back to using the standard "startdate" field that only
+            # supports 2-digit years.
+            # This is what MNE-Python currently parses and puts into
+            # raw.info["meas_date"]
+            if "does not match format '%d-%b-%Y'" not in str(e):
+                raise e
+        daysback, keep_his = _check_anonymize(anonymize, raw, '.edf')
+        info = anonymize_info(raw.info, daysback=daysback, keep_his=keep_his)
+        startdate = datetime.strftime(info['meas_date'], "%d-%b-%Y").upper()
+        meas_date = datetime.strftime(info['meas_date'], "%d.%m.85")
```

Comment:
Please be consistent in the use of single- and double-quotes. Currently, both are used in this function, and I cannot recognise a pattern :)

Reply:
My (potentially weird) personal convention is to use double-quotes for things meant to be used as actual text (i.e. something written to a file or printed out) and single-quotes for everything else (e.g. keys in a dict, arguments to functions). I realize the formatting strings for the dates here don't really match that criteria though, so I'll make them single quotes.